### ISSUE TITLE
Error encountered during make compilation

### ISSUE THREAD

**Issue by WZT666-dev** [fasrc/slurm_showq#15]
[root@mgt slurm_showq-master]# make 
make  all-am
make[1]: Entering directory `/root/slurm/showq/slurm_showq-master'
g++ -DHAVE_CONFIG_H -I.     -g -O2 -MT slurm_showq.o -MD -MP -MF .deps/slurm_showq.Tpo -c -o slurm_showq.o slurm_showq.cpp
slurm_showq.cpp: In member function ‘void Slurm_Showq::query_running_jobs()’:
slurm_showq.cpp:51:14: error: aggregate ‘hostlist_t job_nodelist’ has incomplete type and cannot be defined
   hostlist_t job_nodelist;
              ^
slurm_showq.cpp:54:22: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
   char *delimiters = ",";
                      ^
slurm_showq.cpp:55:17: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
   char *deleq = "=";
                 ^
make[1]: *** [slurm_showq.o] Error 1
make[1]: Leaving directory `/root/slurm/showq/slurm_showq-master'
make: *** [all] Error 2

**Comment by pedmon** [comment]
What compiler and compiler version are you using?

**Comment by james-simone** [comment]
I suspect this is a issue with slurm 23.11.1 since showq compiles with slurm 23.02.6.

`g++ -DHAVE_CONFIG_H -I.     -g -O2 -MT slurm_showq.o -MD -MP -MF .deps/slurm_showq.Tpo -c -o slurm_showq.o slurm_showq.cpp
slurm_showq.cpp: In member function ‘void Slurm_Showq::query_running_jobs()’:
slurm_showq.cpp:51:14: error: aggregate ‘hostlist_t job_nodelist’ has incomplete type and cannot be defined
   hostlist_t job_nodelist;`

**Comment by pedmon** [comment]
Ah okay. We haven't upgraded to 23.11 yet but will be soon. I will need to investigate this when we do.

**Comment by WZT666-dev** [comment]
> What compiler and compiler version are you using?

Sorry, I just saw it. 
Thank you for your reply!
At that time, it was compiled on the Centos7.9 system and used the built-in gcc4.8 error. Later, when changing the version of gcc9.3, it still reported an error. Finally, after changing the version of Slurm, it was able to compile successfully. At first, I used version 23.02.6 of slurm23.02.6, but later switched to version 22.05 of slum22.05. Because this was my attempt in the testing environment, I did not continue exploring.
